packer/all_plugins_storage.go

```diff
@@ -0,0 +1,32 @@
+package packer
+
+import (
+	pluginsdk "github.com/hashicorp/packer-plugin-sdk/plugin"
+)
+
+type PluginComponentType string
+
+const (
+	PluginComponentBuilder       PluginComponentType = "builder"
+	PluginComponentPostProcessor PluginComponentType = "post-processor"
+	PluginComponentProvisioner   PluginComponentType = "provisioner"
+	PluginComponentDataSource    PluginComponentType = "data-source"
+)
+
+type PluginDetails struct {
+	Name        string
+	Description pluginsdk.SetDescription
+	PluginPath  string
+}
+
+var PluginsDetailsStorage = map[string]PluginDetails{}
+
+func AddPluginDetails(
+	componentKey, pluginName, pluginPath string, pluginDescription pluginsdk.SetDescription,
```

Comment:
If the component key is computed outside of this function, do we need it at all? I believe we can populate this map directly at callsite.

If we don't have anything else in here, should we move the map definition to `plugin.go`? This might be more appropriate of a place to keep track of that

Reply:
Alternatively if we want to keep that function, I would still argue in favour of leaving the responsability of creating the `PluginDetails` to the caller instead, that'll make usage a bit simpler in my opinion.

Come to think of it, that plugin detail will be essentially the same for all the components from the same binary, so we could create one instance of the structure, and pass it for each component with the right key?

Reply:
Yeah.. let's move it to the `plugin.go` and get rid of the method.